sync-android/src/main/java/com/cloudant/sync/sqlite/android/EncryptedSavedAttachment.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2014 Cloudant, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.sqlite.android;
+
+import com.cloudant.sync.datastore.Attachment;
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.replication.PushAttachmentsInline;
+import com.cloudant.sync.sqlite.android.encryption.AttachmentStreamFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Logger;
+
+/**
+ * Created by tomblench on 14/03/2014.
+ */
+
+/**
+ * An Attachment which has been retrieved from the Database
+ */
+public class EncryptedSavedAttachment extends Attachment {
```

Comment:
I think these `Encrypted*Attachment` classes are not the right design to follow. At the very least they should inherit as much as possible from a base class. I'll have a think about an alternative and maybe commit it directly.